### PR TITLE
Feature upgrade meteor latest

### PR DESCRIPTION
**Pull request by bogdanrn**
Upgraded meteor and some packages to latest version in order to make the app run locally. 

PS: to make sure you don't have issues, run `rm -rf node_modules && meteor npm i`

There are more changes to be done, but they have to be done by someone who has all the details of the app and infrastructure. 

Some of the remaining changes i noticed: 
- remove unused packages from pacakge.json & upgrade the remaining ones
- change the file structure of the repo because meteor autoloads all files in here (including deploymenet files)

New file structure of the repo should contain only some root folders like: meteor, config, deployement

### CODE REVIEW

**Comment by bogdanrn** [comment]
Because all files from the folder gets loaded to meteor, i had to create some empty `mup-secret.json` files as well.

File: .babelrcdisabled
Comment (carlos-algms) [inline .babelrcdisabled:1]: Maybe delete this file, as it is "disabled"

File: packages/accounts-discourse/.gitignore
Comment (carlos-algms) [inline packages/accounts-discourse/.gitignore:6]:
```suggestion
!package.js
!package-lock.json
```

File: tests/.DS_Store
Comment (carlos-algms) [inline tests/.DS_Store:1]: this file also could be deleted

**Comment by jorgeer** [comment]
Haha, after seeing the post on /r/meteor, I went to do the same thing as I had a little spare time. The changes you have done here are EXACTLY what I did as well, just 30 minutes earlier. Remove webpack, add fourseven:scss, move import structures, rewrite scss imports, replace package-lock, upgrade node-stubs, add babel etc. So I can vouch for the fact that it works and is the right way to go.

File: client/main.html
Comment (carlos-algms) [inline client/main.html:25]: We could maybe include this pre-connect here for a better performance and web-vitals score

```suggestion
  <link rel="preconnect" href="https://fonts.googleapis.com/" crossOrigin="anonymous" />
  <link rel="dns-prefetch" href="https://fonts.googleapis.com/" />
  <link rel="preconnect" href="https://use.fontawesome.com/" crossOrigin="anonymous" />
  <link rel="dns-prefetch" href="https://use.fontawesome.com/" />
  
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.0.13/css/all.css"
```

File: .DS_Store
Comment (carlos-algms) [inline .DS_Store:1]: We could also delete this file. 
How come the MacOs files ended up in the repo if you put `/*` in the git ignore? 
Maybe also put `/.*` there? so it ignores dotfiles?

File: client/main.js
Comment (carlos-algms) [inline client/main.js:21]:
```suggestion
  ReactDOM.render(<App />, rootDiv);
```

File: client/main.html
Comment (rivertam) [reply client/main.html:25]: ^ this suggestion should move the `<!-- FontAwesome Icons -->` line too

File: .DS_Store
Comment (rivertam) [reply .DS_Store:1]: Putting .DS_Store in the top level .gitignore is also pretty good practice

**Comment by AndyatFocallocal** [comment]
This is fantastic thank you @bogdanrn @jorgeer @carlos-algms. 

I think its best to wait for @TLMcNulty to review before merging as he's been through this process before and should have some thoughts on it. You are very welcome to keep discussing and coding here.

**Comment by AndyatFocallocal** [comment]
@bogdanrn @jorgeer @carlos-algms and @rivertam @Marvelxy 

Some others have suggested we add Docker to the app to make it more stable and help users to set up. What are your thoughts, and if its a solution can we also do that?

**Comment by Marvelxy** [comment]
> @bogdanrn @jorgeer @carlos-algms and @rivertam @Marvelxy
> 
> Some others have suggested we add Docker to the app to make it more stable and help users to set up. What are your thoughts, and if its a solution can we also do that?



> @bogdanrn @jorgeer @carlos-algms and @rivertam @Marvelxy
> 
> Some others have suggested we add Docker to the app to make it more stable and help users to set up. What are your thoughts, and if its a solution can we also do that?

I agree! Dockerizing the project is the first step to solving most of the issues we've been facing. Everyone would have the same environment.

**Comment by jorgeer** [comment]
Well.. in my experience, adding more technology/complexity in order to simplify a project can do more harm than good sometimes. Now you have two layers to maintain instead of one, and have to teach new devs to use docker properly in addition to Meteor.

File: .DS_Store
Comment (TLMcNulty) [reply .DS_Store:1]: .DS_Stores should never make it into version control. +1

**Comment by TLMcNulty** [comment]
> Well.. in my experience, adding more technology/complexity in order to simplify a project can do more harm than good sometimes. Now you have two layers to maintain instead of one, and have to teach new devs to use docker properly in addition to Meteor.

Agree. Meteor doesn't do exactly what docker does, but it does attempt enough that putting one inside the other is pretty complex and makes a lot either redundant or unnecessary. 

It's very attractive to me because of the ease of environment setup/portability, but we don't have any build or deployment infrastructure that depends or could benefit from containerization. 

Let's keep it simple. A cleaner and more up to date meteor configuration is a huge step forward in brushing off some cobwebs. I can't check this branch out for validation at the moment but I'll put it together this evening.

**Comment by TLMcNulty** [comment]
I was able to build this in a few clean environments. Let's see how it builds on merge.

**Review by TLMcNulty — APPROVED** [review]

**Comment by AndyatFocallocal** [comment]
Travis is building PHM! I'll let you know if its running in around 10mins. 

@Marvelxy or @TLMcNulty it would be really cool if you could have a little look at deploy-phm as it has some differences to master and i'm not sure where those have come from. It looks to me like PHM-deploy (the live site) hasn't yet received the Meteor update.

**Comment by AndyatFocallocal** [comment]
Although i can see them here: https://github.com/focallocal/fl-maps/commit/df4805463d5f9e176f90ed2bc0a83236217fac68

**Comment by AndyatFocallocal** [comment]
Sorry everyone, i posted on the other PR by mistake so most of you probably missed the update. Your kind and excellent work is greatly appreciated and a big step forwards. There is an error or errors preventing Travis from building the updated platform. Here's the error logs:

Node.js: 16.7.0 [temp.txt](https://github.com/focallocal/fl-maps/files/11186021/temp.txt)

and Node.js: 8.11.1 [temp2.txt](https://github.com/focallocal/fl-maps/files/11186022/temp2.txt)

File: tests/.DS_Store
Comment (bogdanrn) [reply tests/.DS_Store:1]: my bad, didn't touch the gitignore at all

**Comment by bogdanrn** [comment]
For second log, looks like travis is not configured to use the min required node version. Check here https://docs.meteor.com/install.html#prereqs-node

> Node.js version >= 10 and <= 14 is required.

But overall, from the logs, it seems you're using `mup` for deployment. You would have to update the mup config to use a newer docker image. 

Something like
```  
    dockerImage: "zodern/meteor:0.7.0",

    //optional
    docker: {
      imagePort: MUP_CONFIG.APP_PORT,
    },
```

https://github.com/focallocal/fl-maps/blob/554333384be59638064e7b574cb103618ba93d99/deployment/publichappinessmovement/mup.js#L15

https://github.com/focallocal/fl-maps/blob/ea6c71a1a1d24440f1be493d88a9c4e681203708/deployment/testing/mup.js#L13

https://github.com/focallocal/fl-maps/blob/ea6c71a1a1d24440f1be493d88a9c4e681203708/deployment/brightertomorrowmap/mup.js#L15

**Comment by AndyatFocallocal** [comment]
That seems to tie in with my line of research. The eventual solution to our platform being down for over a year after using the auto-migration tool from mLab to MongoDB, was to set the node version to 2.014 or earlier.

Do you think we can update both together, or is that too likely to add complications?

image

**Comment by bogdanrn** [comment]
Do the updates one after another if possible, this way you at least know what broke it. Using later versions of mongodb should not give any errors and even if it does, there's nothing that can't be fixed. 

**Important: Make sure you do your backups before attempting anything.**

**Comment by bogdanrn** [comment]
Btw, looking at the mup.js file, it looks outdated. Make sure you use the correct options for it as well: https://meteor-up.com/docs.html

**Comment by Marvelxy** [comment]
@bogdanrn , could you test this out and open a PR?

**Comment by AndyatFocallocal** [comment]
Any updates on the mup.js fix anyone?

I have some good news on this fix. One of the new devs who joined our team was able to use it to get localhost running, so we are definitely on the right path. Here's their post with how they got it running: https://publichappinessmovement.com/t/topic/1589/49

and their earlier message with the issues they were having and how they solved them:

I finally fixed all the errors! They were ALL due to versions being incompatible - especially sass, node-sass, sass-loader, webpack… I ended up using a website called https://www.npmpeer.dev/ to find compatible versions.

**Comment by TLMcNulty** [comment]
The meteor version in the mup file is a good find. That file is encrypted, you can read about the [implementation of sops here](https://github.com/focallocal/fl-maps/blob/master/deployment/secrets.md).

I have the key with me but don't even have sops installed in this environment so I'll get cracking on decoding that file and re-encrypting. I went through settings.json and mup.js prior to the upgrade merge but didn't look for/pay attention to versions.

**Comment by TLMcNulty** [comment]
Luckily that's not actually an encrypted file. I'll have changes for node-12-base shortly.

**Comment by AndyatFocallocal** [comment]
Making progress but still not building I'm afraid. 

Here's the latest update, can anyone see what's causing the issue?

https://github.com/focallocal/fl-maps/pull/1166#issuecomment-1507699911

We also have a Dev who's working on upgrading React to the latest version over on PHM. Perhaps that's helpful to the Meteor upgrade?

@bogdanrn you said you wanted to give a month to getting this project running smoothly, would it be ok to onboard you to our main site and show you around a bit, then talk about other tasks you could work on once this update is running smoothly.

**Pull request merged**
